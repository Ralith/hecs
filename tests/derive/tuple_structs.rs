use hecs::Bundle;

#[derive(Bundle)]
struct Foo(i32);

#[derive(Bundle)]
struct Bar(i32, String);

#[derive(Bundle)]
struct Baz(i32, String, &'static str);

fn main() {}
