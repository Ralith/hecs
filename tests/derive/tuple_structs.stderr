thread 'rustc' panicked at '`"0"` is not a valid identifier', compiler\rustc_expand\src\proc_macro_server.rs:332:13
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.48.0-nightly (154f1f544 2020-10-02) running on x86_64-pc-windows-msvc

note: compiler flags: -C embed-bitcode=no -C debuginfo=2 -C incremental --crate-type bin

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: proc-macro derive panicked
 --> $DIR/tuple_structs.rs:3:10
  |
3 | #[derive(Bundle)]
  |          ^^^^^^
  |
  = help: message: `"0"` is not a valid identifier

thread 'rustc' panicked at '`"0"` is not a valid identifier', compiler\rustc_expand\src\proc_macro_server.rs:332:13
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.48.0-nightly (154f1f544 2020-10-02) running on x86_64-pc-windows-msvc

note: compiler flags: -C embed-bitcode=no -C debuginfo=2 -C incremental --crate-type bin

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: proc-macro derive panicked
 --> $DIR/tuple_structs.rs:6:10
  |
6 | #[derive(Bundle)]
  |          ^^^^^^
  |
  = help: message: `"0"` is not a valid identifier

thread 'rustc' panicked at '`"0"` is not a valid identifier', compiler\rustc_expand\src\proc_macro_server.rs:332:13
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.48.0-nightly (154f1f544 2020-10-02) running on x86_64-pc-windows-msvc

note: compiler flags: -C embed-bitcode=no -C debuginfo=2 -C incremental --crate-type bin

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: proc-macro derive panicked
 --> $DIR/tuple_structs.rs:9:10
  |
9 | #[derive(Bundle)]
  |          ^^^^^^
  |
  = help: message: `"0"` is not a valid identifier
